src/comm/TcpAcceptor.cc

```diff
@@ -323,10 +324,14 @@ Comm::TcpAcceptor::notify(const Comm::Flag flag, const Comm::ConnectionPointer &
         params.xaction = new MasterXaction(XactionInitiator::initClient);
         params.xaction->squidPort = listenPort_;
         params.fd = conn->fd;
+        conn->enterOrphanage();
         params.conn = params.xaction->tcpClient = newConnDetails;
         params.flag = flag;
         params.xerrno = errcode;
         ScheduleCallHere(call);
+    } else {
+        // disconnect clients we cannot respond to
+        conn->close();
```

Comment:
This will close the _listening_ connection.

Reply:
Fixed in b4e50a8